sdk/tests/client/input_selection/burn.rs

```diff
@@ -885,3 +885,59 @@ fn burn_native_tokens() {
         Some(vec![(TOKEN_ID_1, 80), (TOKEN_ID_2, 70)])
     ));
 }
+
+#[test]
+fn burn_foundry_and_its_alias() {
+    let protocol_parameters = protocol_parameters();
+    let alias_id_1 = AliasId::from_str(ALIAS_ID_1).unwrap();
+
+    let inputs = build_inputs(vec![
+        Foundry(
+            1_000_000,
+            alias_id_1,
+            0,
+            SimpleTokenScheme::new(U256::from(0), U256::from(0), U256::from(10)).unwrap(),
+            None,
+        ),
+        Alias(
+            1_000_000,
+            alias_id_1,
+            0,
+            BECH32_ADDRESS_ED25519_0,
+            BECH32_ADDRESS_ED25519_0,
+            None,
+            None,
+            None,
+            None,
+        ),
```

Comment:
Should be rather like this, because the with_foundry_counter needs to be set
```
        AliasOutputBuilder::new_with_amount(1_000_000, alias_id_1)
            .add_unlock_condition(StateControllerAddressUnlockCondition::new(
                Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
            ))
            .add_unlock_condition(GovernorAddressUnlockCondition::new(
                Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
            ))
            .with_state_index(1)
            .with_foundry_counter(1)
            .finish_output(protocol_parameters.token_supply())
            .unwrap(),
```

Reply:
Yeah right, maybe I should have just pushed it to your branch

Reply:
Looks like it passes without this in your PR